src/lib.rs

```diff
@@ -1079,6 +1138,7 @@ macro_rules! flags {
     ($(#[$m:meta])* $p:vis enum $n:ident: $t:ty { $($(#[$a:meta])*$k:ident = $v:expr),+ $(,)* } $($next:tt)*) => {
         $(#[$m])*
         #[derive(Copy, Clone, Debug, PartialEq, Eq)]
+        #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
```

Comment:
Actually, I'm not too sure we should be doing this cause flags will typically have to be serialized as their `repr` size. So if we do want to implement Serialize/Deserialize for the enum, we should do it manually, like we do for `FlagSet<T>` above.

Reply:
@npmccallum WDYT?

Reply:
After trying to implement it, I realized it's going to be more work than worth it. See my latest patch that could be squashed into this one, once you approve it.

Reply:
I see the following options:

1. Don't serialize the flags at all, just `FlagSet`.
2. Derive on the flags enum directly (with or without repr).
3. Manually implement ser/de by internally converting to/from `FlagSet`.

How common do you think it will be that people need to ser/de on the flags directly?

Reply:
@npmccallum I don't see it being common or even needed really so that's why I went for option#1 with docs describing how one can easily implement ser/de for flags enum.

Reply:
Okay. That plan sounds reasonable. Please refactor the commits to a logical set of changes and I'll finish the review and merge.